tests/integration/test_sources.py

```
@@ -970,6 +970,112 @@ async def test_get_fulltext_empty_content(
+        async with NotebookLMClient(auth_tokens) as client:
+            await client.sources.get_fulltext("nb_123", "src_x", output_format="markdown")
+
+        body = urllib.parse.unquote(httpx_mock.get_request().content.decode())
+        assert "[3]" in body
+        assert "[2]" not in body or body.count("[3]") >= 2
+
+    @pytest.mark.asyncio
+    async def test_get_fulltext_invalid_format_raises(self, auth_tokens):
+        """Unknown output_format must fail fast before any RPC call."""
+        async with NotebookLMClient(auth_tokens) as client:
+            with pytest.raises(ValueError, match="text.*markdown"):
```

Ruff (0.15.12)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)